components/edit_post_modal/edit_post_modal.jsx

```diff
@@ -0,0 +1,337 @@
+        if (this.state.postError !== postError) {
+            this.setState({postError});
+        }
+    }
+
+    handleEdit = async () => {
+        const updatedPost = {
+            message: this.state.editText,
+            id: this.props.editingPost.postId,
+            channel_id: this.props.editingPost.post.channel_id
+        };
+
+        if (this.state.postError) {
+            this.setState({errorClass: 'animation--highlight'});
+            setTimeout(() => {
+                this.setState({errorClass: null});
+            }, Constants.ANIMATION_TIMEOUT);
+            return;
+        }
+
+        if (updatedPost.message === this.props.editingPost.post.message) {
+            // no changes so just close the modal
+            this.handleHide();
+            return;
+        }
+
+        this.props.actions.addMessageIntoHistory(updatedPost.message);
+
+        if (updatedPost.message.trim().length === 0) {
+            this.handleHide();
+            GlobalActions.showDeletePostModal(Selectors.getPost(getState(), this.props.editingPost.postId), this.props.editingPost.commentsCount);
+            return;
+        }
+
+        const data = await this.props.actions.editPost(updatedPost);
+        if (data) {
+            window.scrollTo(0, 0);
+        }
+
+        this.handleHide();
+    }
+
+    handleChange = (e) => {
+        const message = e.target.value;
+        this.setState({
+            editText: message
+        });
+    }
+
+    handleEditKeyPress = (e) => {
+        if (!UserAgent.isMobile() && !this.props.ctrlSend && e.which === KeyCodes.ENTER && !e.shiftKey && !e.altKey) {
+            e.preventDefault();
+            ReactDOM.findDOMNode(this.refs.editbox).blur();
+            this.handleEdit();
+        } else if (this.props.ctrlSend && e.ctrlKey && e.which === KeyCodes.ENTER) {
+            e.preventDefault();
+            ReactDOM.findDOMNode(this.refs.editbox).blur();
+            this.handleEdit();
+        }
+    }
+
+    handleKeyDown = (e) => {
+        if (this.props.ctrlSend && e.keyCode === KeyCodes.ENTER && e.ctrlKey === true) {
+            this.handleEdit();
+        }
+    }
+
+    handleHide = () => {
+        this.setState({hidding: true});
+    }
+
+    handleEnter = () => {
+        this.setState({
+            editText: this.props.editingPost.post.message,
+            error: '',
+            typing: false
+        });
+    }
+
+    handleEntered = () => {
+        this.refs.editbox.focus();
+        this.refs.editbox.recalculateSize();
+    }
+
+    handleExit = () => {
+        this.refs.editbox.hidePreview();
+    }
+
+    handleExited = () => {
+        const refocusId = this.props.editingPost.refocusId;
+        if (refocusId !== '') {
+            setTimeout(() => {
+                const element = $(refocusId).get(0);
+                if (element) {
+                    element.focus();
+                }
+            });
+        }
+        this.props.actions.setEditingPost('', 0, '', '');
```

Comment:
0/5, maybe set defaults on the action so you can just do `this.props.actions.setEditingPost()` here

Reply:
Yes, I agree, the default values could be no-post.